chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -358,29 +399,44 @@ impl NetworkState {
                 return false;
             }
         }
-        match self.graph.routing_table.find_route(&clock, &msg.target) {
-            Ok(peer_id) => {
-                // Remember if we expect a response for this message.
-                if msg.author == my_peer_id && msg.expect_response() {
-                    tracing::trace!(target: "network", ?msg, "initiate route back");
-                    self.graph.routing_table.add_route_back(&clock, msg.hash(), my_peer_id);
-                }
-                return self.tier2.send_message(peer_id, Arc::new(PeerMessage::Routed(msg)));
+        match tier {
+            tcp::Tier::T1 => {
+                tracing::debug!(target:"test", "sending msg over TIER1");
+                let peer_id = match &msg.target {
+                    PeerIdOrHash::Hash(hash) => {
+                        match self.tier1_route_back.lock().remove(clock, hash) {
+                            Some(peer_id) => peer_id,
+                            None => return false,
+                        }
+                    }
+                    PeerIdOrHash::PeerId(peer_id) => peer_id.clone(),
+                };
+                return self.tier1.send_message(peer_id, Arc::new(PeerMessage::Routed(msg)));
             }
-            Err(find_route_error) => {
-                // TODO(MarX, #1369): Message is dropped here. Define policy for this case.
-                metrics::MessageDropped::NoRouteFound.inc(&msg.body);
+            tcp::Tier::T2 => match self.graph.routing_table.find_route(&clock, &msg.target) {
+                Ok(peer_id) => {
+                    // Remember if we expect a response for this message.
+                    if msg.author == my_peer_id && msg.expect_response() {
+                        tracing::trace!(target: "network", ?msg, "initiate route back");
+                        self.graph.routing_table.add_route_back(&clock, msg.hash(), my_peer_id);
```

Comment:
side comment: I actually wonder - why do we do that? (if we expect the response, it should simply arrive and be addressed to us. Why do we add anything to the routing table..)